cpp/src/arrow/ipc/metadata-internal.cc

```diff
@@ -246,6 +246,16 @@ Status FieldFromFlatbuffer(const flatbuf::Field* field, std::shared_ptr<Field>*
 
 // Implement MessageBuilder
 
+// will return the endianness of the system we are running on
+flatbuf::Endianness endianness() {
+  union {
+    uint32_t i;
+    char c[4];
+  } bint = {0x01020304};
+
+  return bint.c[0] == 1 ? flatbuf::Endianness_Big : flatbuf::Endianness_Little;
```

Comment:
Yes, that's where it's from I guess since this was a recommended approach by 
http://stackoverflow.com/users/11465/david-cournapeau
http://stackoverflow.com/a/1001373/5492075
Although there's little variation in how many ways you can determine endianness.
Should I add a comment with attribution?